backend/src/test/java/fr/anettehaferkorn/backend/service/MatchingServiceTest.java

```diff
@@ -0,0 +1,125 @@
+package fr.anettehaferkorn.backend.service;
+import fr.anettehaferkorn.backend.controller.exception.NoMatchingWineException;
+import fr.anettehaferkorn.backend.model.RecommendationDTO;
+import fr.anettehaferkorn.backend.model.WineGrape;
+import fr.anettehaferkorn.backend.model.WineQuery;
+import fr.anettehaferkorn.backend.repo.WineGrapeRepository;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class MatchingServiceTest {
+     WineGrapeRepository wineGrapeRepository= mock(WineGrapeRepository.class);
+
+    @Test
+    void getMatchingWines() {
+        MatchingService matchingService = new MatchingService(wineGrapeRepository);
+        WineQuery wineQuery=new WineQuery("dinner","red","France","light","sweet");
+        List<WineGrape>mockedRecommendations=List.of(
+                new WineGrape("Merlot","dinner","red","France","light","sweet","abcd"),
+                new WineGrape("BBBB","dinner","red","France","strong","dry","abcd"),
+                new WineGrape("CCCC","dinner","red","France","strong","sweet","abcd")
+                );
+        List<RecommendationDTO>mockedMatches=List.of(
+                new RecommendationDTO("Merlot","dinner","red","France","light","sweet","abcd",3),
+                new RecommendationDTO("CCCC","dinner","red","France","strong","sweet","abcd",2)
+        );
+
+       when(wineGrapeRepository.findWineQueryDTOByOccasion(wineQuery.getOccasion())).thenReturn(mockedRecommendations);
+
+        //When
+        List<RecommendationDTO> actual = matchingService.getMatchingWines(wineQuery);
+        //Then
+        assertEquals(mockedMatches, actual);
+
+    }
+
+    @Test
+    void getMatchingWines_2() {
```

Comment:
find a better name that describes what this test is doing differently than the first one